xbmc/addons/interfaces/Addon/AddonCallbacksAddon.cpp

```diff
@@ -513,10 +514,37 @@ char *CAddonCallbacksAddon::GetFileProperty(const void* addonData, void* file, X
 
   CFile* cfile = (CFile*)file;
   if (cfile)
-    return strdup(cfile->GetProperty(type, name).c_str());
+  {
+    std::vector<std::string> values = cfile->GetPropertyValues(type, name);
+    if (values.empty()) {
+      return nullptr;
+    }
+    strdup(values[0].c_str());
+  }
   return nullptr;
 }
 
+char **CAddonCallbacksAddon::GetFilePropertyValues(const void* addonData, void* file, XFILE::FileProperty type, const char *name, int *numValues)
+{
+  CAddonInterfaces* helper = (CAddonInterfaces*)addonData;
+  if (!helper)
+    return nullptr;
+
+  CFile* cfile = (CFile*)file;
```

Comment:
Please use c++ style casts - here and everywhere in the code your PR introduces. Yes, we use c style casts in old code, but should use only c++ style casts in new code.